Modules/TPC/include/TPC/TH1ReductorTPC.h

```diff
@@ -0,0 +1,56 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file   TH1ReductorTPC.h
+/// \author Marcel Lesch
+/// \author Cindy Mordasini
+/// \author Based on the work from Piotr Konopka
+///
+
+#ifndef QUALITYCONTROL_TH1REDUCTORTPC_H
+#define QUALITYCONTROL_TH1REDUCTORTPC_H
+
+#include "TPC/ReductorTPC.h"
+#include <vector>
+
+namespace o2::quality_control_modules::tpc
+{
+
+/// \brief A TPC-specific reductor which obtains the most popular characteristics of TH1.
+///
+/// A Reductor which obtains the most popular characteristics of TH1, in the case of TPC-related quantities.
+/// It produces a branch in the format: "mean/D:stddev:entries". All these variables are saved as arrays of doubles with a maximum length of 72 (i.e. number of ROCs in the full TPC).
+class TH1ReductorTPC : public quality_control_modules::tpc::ReductorTPC
+{
+ public:
+  /// \brief Constructor.
+  TH1ReductorTPC() = default;
+  /// \brief Destructor.
+  ~TH1ReductorTPC() = default;
+
+  /// \brief Definitions of the methods common to all reductors.
+  void* getBranchAddress() override;
+  const char* getBranchLeafList() override;
+  void update(TObject* obj, std::vector<std::vector<float>>& axis, bool isCanvas) override;
+
+ private:
+  static constexpr int NMAXSLICES = 72; ///< Maximum number of slices, or histograms for an input canvas. TBI: Make configurable.
+
+  struct {
+    Double_t mean[NMAXSLICES];    ///< Mean value for each slice/histogram.
+    Double_t stddev[NMAXSLICES];  ///< Standard deviation for each slice/histogram.
+    Double_t entries[NMAXSLICES]; ///< Number of entries for each slice/histogram.
+  } mStats;
```

Comment:
Thank you! Maybe best would be to implement all the other changes in the draft, and focus on this once the PR is done?

Reply:
Not sure I understand. I would clean this PR up before merging, if this is what you mean.

Reply:
Yes, I would also take care of it now. I am not sure, if the second option is necessary, but @cimordas didn't you try something like the first one? In any case, should you implement the 2nd one, I also think that all the arrays should have the same size, at least I don't see any reason why the shouldn't.

Reply:
Quick update on this one. I tried to implement a variation of the first proposition, with double instead of arrays of double inside the struct. However, QC does not build and returns the following error (the full log is attached):
```
Modules/TPC/G__O2QcTPC.cxx: In function 'ROOT::TGenericClassInfo* ROOT::GenerateInitInstanceLocal(const std::vector<o2::quality_control_modules::tpc::TH2ReductorTPC::SliceInfo>*)':
Modules/TPC/G__O2QcTPC.cxx:1377:62: error: 'struct o2::quality_control_modules::tpc::TH2ReductorTPC::SliceInfo' is private within this context
 1377 |    static TGenericClassInfo *GenerateInitInstanceLocal(const vector<o2::quality_control_modules::tpc::TH2ReductorTPC::SliceInfo>*)
      |                                                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from Modules/TPC/G__O2QcTPC.cxx:52:
/home/cindy/aliceO2/sw/SOURCES/QualityControl/master/0/Modules/TPC/include/TPC/TH2ReductorTPC.h:45:10: note: declared private here
   45 |   struct SliceInfo {  ///< All quantities of TH2 for each x-slice/histogram.
      |          ^~~~~~~~~
```

@wiechula Do you have an idea what could be the problem?
[log.txt](https://github.com/AliceO2Group/QualityControl/files/6916394/log.txt)